mate.mate-themes: 3.22.20 -> 3.22.21

romildo opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

Update to version [3.22.21](https://github.com/mate-desktop/mate-themes/releases/tag/v3.22.21)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

romildo committed:
mate.mate-themes: 3.22.20 -> 3.22.21

jonringer reviewed (approved):
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/80925
1 package built:
mate.mate-themes
```

closure size is the same.

worldofpeace commented on pkgs/desktops/mate/mate-themes/default.nix:
```diff
@@ -1,18 +1,18 @@
+    sha256 = "051g2vq817g84yrqzf7hjcqr4xrghnw1rprjd6jf5mhhzmwcas6n";
   };
 
   nativeBuildInputs = [ pkgconfig intltool gtk3 ];
```
```suggestion
  nativeBuildInputs = [ pkgconfig gettext gtk3 ];
```

migrate from intltool to gettext https://github.com/mate-desktop/mate-themes/releases/tag/v3.22.21

romildo replied:
Done.

worldofpeace reviewed (changes requested):
I see there's some icon themes here
```
ContrastHigh  mate  mate-black
```

one of which`ContrastHigh` you generate an icon cache for.
You'll need to set `dontDropIconThemeCache = true;` so the gtk setup hook doesn't remove it.

Should we generate icon cache's for the other icon themes as well?

romildo committed:
mate.mate-themes: migrate from intltool to gettext

romildo committed:
mate.mate-themes: don't drop icon caches

Set dontDropIconThemeCache so the gtk setup hook doesn't remove the
icon cache.

romildo commented:
> I see there's some icon themes here
> 
> ```
> ContrastHigh  mate  mate-black
> ```
> 
> one of which`ContrastHigh` you generate an icon cache for.
> You'll need to set `dontDropIconThemeCache = true;` so the gtk setup hook doesn't remove it.

Done.

> Should we generate icon cache's for the other icon themes as well?

I suppose no, because they are cursors only.
